ofp/packet.go

```diff
@@ -179,5 +182,14 @@ func (p *PacketOut) ReadFrom(r io.Reader) (int64, error) {
 	p.Actions = nil
 
 	nn, err := p.Actions.ReadFrom(limrd)
-	return n + nn, err
+	if err != nil {
+		return n + nn, err
+	}
+
+	p.Data, err = ioutil.ReadAll(r)
+	if err != nil {
+		// return n + nn + int64(len(p.Data)), err
```

Comment:
len(p.Data) panics with invalid memory address or nil pointer dereference when Packet Out has no Data (first test-case).
on the other hand, a check like "if data == nil" { return ...} never worked

Reply:
Could you, please, try the following implementation of ```ReadFrom``` method (tests are passing on my local setup):
```go
func (p *PacketOut) ReadFrom(r io.Reader) (int64, error) {
        var plen uint16

        n, err := encoding.ReadFrom(r, &p.Buffer, &p.InPort,
                &plen, &defaultPad6)

        if err != nil {
                return n, err
        }

        limrd := io.LimitReader(r, int64(plen))
        p.Actions = nil

        nn, err := p.Actions.ReadFrom(limrd)
        if n += nn; err != nil {
                return n, err
        }

        p.Data, err = ioutil.ReadAll(r)
        return n + int64(len(p.Data)), err
}
```

I don't know why ```len``` leads to a panic, my code is almost the same except of some clean-up. What version of go are you using? Do you use something extravagant as ```gccgo```?